### PR TITLE
KAFKA-3691: Confusing logging during metadata update timeout

### PR DESCRIPTION
**Pull request by granthenke**

### CODE REVIEW

File: clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java [hunk 1]
```diff
@@ -522,7 +522,12 @@ private long waitOnMetadata(String topic, long maxWaitMs) throws InterruptedExce
             log.trace("Requesting metadata update for topic {}.", topic);
             int version = metadata.requestUpdate();
             sender.wakeup();
-            metadata.awaitUpdate(version, remainingWaitMs);
+            try {
+                metadata.awaitUpdate(version, remainingWaitMs);
```
Comment (ijuma) [inline clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java:526]: Is it actually useful for `metadata.awaitUpdate` to throw an exception? Maybe it should simply return a `boolean`.

Comment (granthenke) [inline clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java:526]: That could be a useful change. It is a public method though and I am not sure its valuable enough of a change to deal with any compatibility concerns.

Comment (ijuma) [inline clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java:526]: It's a public method, true, but `Metadata` is not considered public API (it's annoying that we don't make this crystal clear via package names). Still, you're probably right that it's not worth potentially breaking people for this small improvement.

**Comment by ijuma** [comment]
LGTM

**Pull request closed without merging**